From 2662860491ff44a16e01581a6b3727d53997fee8 Mon Sep 17 00:00:00 2001
From: Pablo Hernandez-Cerdan <pablo.hernandez.cerdan@outlook.com>
Date: Tue, 15 May 2018 16:17:01 -0400
Subject: [PATCH] COMP: Add gcc8 support to vcl_compiler.

It just adds a if/else branch to the existing logic.

Triggered here: https://discourse.itk.org/t/building-and-compiling-with-gcc-8-1/934/4
---
 vcl/vcl_compiler.h | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/vcl/vcl_compiler.h b/vcl/vcl_compiler.h
index 6a3f2e2152..5ca624603d 100644
--- a/vcl/vcl_compiler.h
+++ b/vcl/vcl_compiler.h
@@ -86,6 +86,17 @@
 #  else
 #   define VCL_GCC_70
 #  endif
+# elif (__GNUC__==8)
+#  define VCL_GCC_8
+#  if (__GNUC_MINOR__ > 2 )
+#   define VCL_GCC_83
+#  elif (__GNUC_MINOR__ > 1 )
+#   define VCL_GCC_82
+#  elif (__GNUC_MINOR__ > 0 )
+#   define VCL_GCC_81
+#  else
+#   define VCL_GCC_80
+#  endif
 # else
 #  error "Dunno about this gcc"
 # endif
